File: ABC262/B_Triangle(easier)/262B220802.cpp
#include <iostream>
#include <vector>

using namespace std;



int main() {
    int n, m;
    int u, v;
    int ans = 0;
    cin >> n >> m;
    vector attached(n, vector<bool>(n));
    
    for(int i = 0 ; i < m ; i++){
        cin >> u >> v;
        attached[u-1][v-1] = true;
        attached[v-1][u-1] = true;
    }
    
    for(int a = 0; a < n ; a++){
        for(int b = a + 1 ; b < n; b++){
            for(int c = b + 1 ; c < n; c++){
                if (attached[a][b] && attached[b][c] && attached[c][a])
                    ans++;
            }
        }
    }
    
    cout << ans << endl;
    
    return 0;
}

File: ABC262/B_Triangle(easier)/262B220802.py

def main():
    n, m = map(int, input().split())
    attached = [[False] * n for _ in range(n)]
    ans = 0

    for _ in range(m): 
        u, v = map(int, input().split())
        attached[u-1][v-1] = True 
        attached[v-1][u-1] = True 

    for a in range(n):
        for b in range(a + 1, n):
            for c in range(b + 1, n):
                if attached[a][b] and attached[b][c] and attached[c][a]:
                    ans += 1

    print(ans)

    


if __name__ == "__main__":
    main() 
